### ISSUE TITLE
Finding coresponding error line in CSV file

### ISSUE THREAD

**Issue by jmarkham910** [adaltas/node-csv-parse#217]
The following code running against Node 8.11, used parser.lines to report the corresponding line that was in error in the original CSV file, (i.e. "Line 6: First name is missing"). I need the ability to get the same information running csv-parse with Node 10. In node 10, parser.lines is usually set to the last line for the CSV by the time parser.on("readable", ...) is executed.
Our CSV has comment lines at the top of the file to explain a few fields to our customers. The customer can always delete the comment lines if they choose. Its not as simple as using a accumulator to count the records in parser.on("readable", ...), as comment lines are not parsed.
```
const parserOptions = {
        trim: true,
        columns: true,
        comment: "#",
        relax_column_count: true,
        skip_empty_lines: true,
        rowDelimiter: ["\r\n", "\n", "\r"]
    };
const parser = require("csv-parse")(parserOptions);

const errorList = [];

parser.on("readable", () => {
    while ((record = parser.read()) !== null) { 
        let errStr = "Line " + parser.lines + ": ",
            noErr = true;
        if (!record[firstName]) {
            noError = false;
            errorStr += "First name is missing";
            errorList.push(errorStr);
        }
        ...
        if (noError) {
            transformer.write(record);
        }
    }
});
```

**Comment by wdavidw** [comment]
Well, if you take a look at the recent commit, I am currently finishing a complete rewrite of the parser. Beside re-implementing entirely the parser using Buffer for performance reasons, I also spent some time (I was working on it today) fixing the line implementation which was not correctly implemented in earlier version (at least when the row delimiter was not the same than the record delimiter. Thus, in the new version, the line property shall be much more reliable.

I am not sure this will answer your issue. I encourage you to look at the code present in the master branch and, if it does solve your problem, to prepare me a small dataset illustrating what you wish to get.

**Comment by jmarkham910** [comment]
Looked at master, and I assume you're referring to the "info" property. I don't think it is helpful for the situation I'm referring to. I want to know which specific line from the CSV corresponds to the record that parser.read returns. Knowing that the CSV contains 2 comment lines, does not tell me the current record's relative position in the file.
I need to be able to tell the customer that line 5 of the CSV below, is missing the required FirstName entry.
```
FirstName,LastName,Email
# Comment line 1
# Comment line 2
Fred,Jacob,fred.jacob@blah.com
,Anderson,alan.anderson@blah.com
Jack,Phillips,jack.phillips@blah.com
```
Perhaps if the returned record also returned metadata:
```
record = {
    FirstName: "",
    LastName: "Anderson",
    Email: "alan.anderson@blah.com",
    metadata: {
        csvLine: 5,
        numberOfFields: 3,
        expectedNumberOfFields: 3
        ... other data specific to this record
    }
```

**Comment by wdavidw** [comment]
I have just created a new `info` option which works similarly to the `raw` option. It shall feet your needs. You can either use the latest code on GitHub or wait for a couple of days until version 4 released. You can see it in action in the [test](https://github.com/adaltas/node-csv-parse/blob/master/test/options.info.coffee)

**Comment by wdavidw** [comment]
Thank you for reporting the issue, it is fixed with 4.0.1.
